mason.cmake

```diff
@@ -0,0 +1,216 @@
+string(RANDOM LENGTH 16 MASON_INVOCATION)
+
+# Directory where Mason packages are located; typically ends with mason_packages
+if (NOT MASON_PACKAGE_DIR)
+    set(MASON_PACKAGE_DIR "${CMAKE_SOURCE_DIR}/mason_packages")
+endif()
+
+# URL prefix of where packages are located.
+if (NOT MASON_REPOSITORY)
+    set(MASON_REPOSITORY "https://mason-binaries.s3.amazonaws.com")
+endif()
+
+# Path to Mason executable
+if (NOT MASON_COMMAND)
+    set(MASON_COMMAND "${CMAKE_SOURCE_DIR}/.mason/mason")
+endif()
+
+# Determine platform
+if(NOT MASON_PLATFORM)
+    if (CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
+        set(MASON_PLATFORM "macos")
```

Comment:
Yes. We are setting `macos` because we expect that upstream providers/users also use `macos`, but we change it to `osx` a little further down.